prometheus/client.go

```diff
@@ -42,7 +62,8 @@ func NewClient() (*Client, error) {
 // Destination service is not included in the map as it is passed as argument.
 // It returns an error on any problem.
 func (in *Client) GetSourceServices(namespace string, servicename string) (map[string]string, error) {
-	query := fmt.Sprintf(istio_request_query, servicename, namespace)
+	query := fmt.Sprintf("istio_request_count{destination_service=\"%s.%s.%s\"}",
+		servicename, namespace, config.Get().Istio_Identity_Domain)
```

Comment:
Ok, if you introduce the Istio_Identity_Domain I guess that then it makes sense to filter it in the method, a couple of lines  below there is a 
```
			if i := strings.Index(sourceService, ".svc"); i > 0 {
				sourceService = sourceService[:i]
			}
```
that should be updated with the specific variable.